tests/src/upgrade.rs

```diff
@@ -0,0 +1,401 @@
+
+    let install_request = InstallerRequestBuilder::new(*DEFAULT_ACCOUNT_ADDR, CONTRACT_1_0_0_WASM)
+        .with_collection_name(NFT_TEST_COLLECTION.to_string())
+        .with_collection_symbol(NFT_TEST_SYMBOL.to_string())
+        .with_total_token_supply(100u64)
+        .with_ownership_mode(OwnershipMode::Minter)
+        .with_identifier_mode(NFTIdentifierMode::Ordinal)
+        .with_nft_metadata_kind(NFTMetadataKind::Raw)
+        .build();
+
+    builder.exec(install_request).expect_success().commit();
+
+    let nft_contract_hash_1_0_0 = support::get_nft_contract_hash(&builder);
+    let nft_contract_key_1_0_0: Key = nft_contract_hash_1_0_0.into();
+
+    for _ in 0..3 {
+        let mint_request = ExecuteRequestBuilder::standard(
+            *DEFAULT_ACCOUNT_ADDR,
+            MINT_SESSION_WASM,
+            runtime_args! {
+                ARG_NFT_CONTRACT_HASH => nft_contract_key_1_0_0,
+                ARG_TOKEN_OWNER => Key::Account(*DEFAULT_ACCOUNT_ADDR),
+                ARG_TOKEN_META_DATA => "",
+            },
+        )
+        .build();
+
+        builder.exec(mint_request).expect_success().commit();
+    }
+
+    let previous_token_representation = support::get_dictionary_value_from_key::<Vec<u64>>(
+        &builder,
+        &nft_contract_key_1_0_0,
+        OWNED_TOKENS,
+        &DEFAULT_ACCOUNT_ADDR.clone().to_string(),
+    );
+
+    assert_eq!(previous_token_representation, vec![0, 1, 2]);
+
+    let maybe_access_named_key = builder
+        .query(None, Key::Account(*DEFAULT_ACCOUNT_ADDR), &[])
+        .unwrap()
+        .as_account()
+        .unwrap()
+        .named_keys()
+        .get(ACCESS_KEY_NAME)
+        .is_some();
+
+    assert!(maybe_access_named_key);
+
+    let package_hash = support::get_nft_contract_package_hash(&builder);
+
+    let upgrade_request = ExecuteRequestBuilder::standard(
+        *DEFAULT_ACCOUNT_ADDR,
+        NFT_CONTRACT_WASM,
+        runtime_args! {
+            ARG_NFT_PACKAGE_HASH => package_hash
+        },
+    )
+    .build();
+
+    builder.exec(upgrade_request).expect_success().commit();
+
+    let nft_contract_hash = support::get_nft_contract_hash(&builder);
+    let nft_contract_key: Key = nft_contract_hash.into();
+
+    let actual_page_record_width = builder
+        .query(None, nft_contract_key, &[MAX_PAGE_NUMBER.to_string()])
+        .expect("must have the stored value")
+        .as_cl_value()
+        .map(|page_cl_value| CLValue::into_t::<u64>(page_cl_value.clone()))
+        .unwrap()
+        .expect("must convert");
+
+    let expected_page_record_width = 100u64 / PAGE_SIZE;
+
+    assert_eq!(expected_page_record_width, actual_page_record_width);
+
+    let actual_page = support::get_token_page_by_id(
+        &builder,
+        &nft_contract_key,
+        &Key::Account(*DEFAULT_ACCOUNT_ADDR),
+        0u64,
+    );
+
+    let expected_page = {
+        let mut page = vec![false; 10];
+        for page_entry in page.iter_mut().take(3) {
```

Comment:
and also here

Reply:
Uses PAGE_SIZE instead of 10  and the extracted variable